### PR TITLE
Implement APIs for Subscription Instances

### PR DESCRIPTION
**Pull request by TronTram**
# Motivation / Background
- Ticket: https://thepressingly.atlassian.net/browse/PINET-640
- To support the QA team in triggering the finalize process more effectively, we will provide an API endpoint `subscription_instances/finalize/:id`. This allows the process to be triggered immediately instead of waiting until the end of the billing period.

*This Pull Request has been created because:*
Reasons for the change

# Detail

*This Pull Request changes:*

List of things you do in this PR:
- Add serializer for subscription instance and subscription instance item.
- Update subscription serializer to include subscription instances
- Add SubscriptionInstancesController with finalize and show action
- Fix `finalize_service.rb` to use 'message' instead of 'error_message' in service_failure!

# Additional information
- Finalize API:
```curl
curl --location --request POST 'http://localhost:3000/api/v1/subscription_instances/finalize/4d1489ad-faab-4361-a51f-1185f60f49d5' \
--header 'Authorization: Bearer e1c02151-059c-4577-bdac-a67bd36e0c82' \
--header 'Content-Type: application/json' \
--data ''
```

- Update subscription API to include `subscription instance` and `subscription instance item`
image

- API to retrieve a subscription instance:
image

*TIP: Provide additional information such as screenshots, benchmarks, reference to other repositories or alternative solutions*

# Checklist

Before submitting the PR make sure the following are checked:

* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.

### CODE REVIEW

File: config.ru [hunk 1]
```diff
@@ -6,12 +6,12 @@ require_relative 'config/environment'
...
-  s.add_http2_port(ENV['GRPC_HOST'] + ":" + ENV['GRPC_PORT'], :this_port_is_insecure)
-  s.handle(Grpc::Server::AuthTokenService)
-  s.run_till_terminated
-end
+# Thread.new do # rubocop:disable ThreadSafety/NewThread
+#   s = GRPC::RpcServer.new
+#   s.add_http2_port(ENV['GRPC_HOST'] + ":" + ENV['GRPC_PORT'], :this_port_is_insecure)
+#   s.handle(Grpc::Server::AuthTokenService)
+#   s.run_till_terminated
+# end
```
Comment (tnlong311) [inline config.ru:14]: remove unused code

File: config/routes.rb [hunk 2]
```diff
@@ -82,6 +82,11 @@
         get :public_key, on: :collection
         get :json_public_key, on: :collection
       end
+
+      # TODO: only allow in staging and development
```
Comment (tnlong311) [inline config/routes.rb:86]: implement the TODO

File: packs/publisher_portal/app/services/subscription_charges/create_service.rb [hunk 3]
```diff
@@ -26,12 +26,12 @@ def call
       Rails.logger.info("Subcription charge creation payload: #{payload}")
 
       subscription_charge_result = stub.create_subscription_charge(Revenue::CreateSubscriptionChargeReq.new(payload))
-      Rails.logger.info("Subcription charge creation subscription_charge_result: #{payload}")
+      Rails.logger.info("Subcription charge creation subscription_charge_result: #{subscription_charge_result}")
```
Comment (tnlong311) [inline packs/publisher_portal/app/services/subscription_charges/create_service.rb:29]: This message is grammatically weird. How about something like `Create subscription charge result:`

File: config.ru (hunk 1 above)
Comment (TronTram) [reply config.ru:14]: Because the grpc server setup is getting a RunTime error, I comment out to avoid this error

File: config/routes.rb (hunk 2 above)
Comment (TronTram) [reply config/routes.rb:86]: I will implement it later.

**Review by tnlong311 — DISMISSED** [review]

**Pull request merged**